cmd/exec.go

```diff
@@ -15,148 +12,83 @@ import (
 	"github.com/unweave/unweave/api/types"
 )
 
-func gitIsStatusClean() (bool, error) {
-	var stdout, stderr bytes.Buffer
-	cmd := exec.Command("git", "status", "--porcelain")
-	cmd.Stdout = &stdout
-	cmd.Stderr = &stderr
-
-	if err := cmd.Run(); err != nil {
-		return false, fmt.Errorf("failed to get git status: %s %s", stderr.String(), stdout.String())
-	}
-	return strings.TrimSpace(stdout.String()) == "", nil
+func Exec(cmd *cobra.Command, args []string) error {
+	return runSSHConnectionCommand(cmd, args, &execCommandFlow{})
 }
 
-func getGitRemoteURL() (string, error) {
-	var stdout, stderr bytes.Buffer
-	cmd := exec.Command("git", "config", "--get", "remote.origin.url")
-	cmd.Stdout = &stdout
-	cmd.Stderr = &stderr
+type execCommandFlow struct{}
 
-	if err := cmd.Run(); err != nil {
-		return "", fmt.Errorf("failed to get git remote URL: %s", stderr.String())
+func (e *execCommandFlow) parseArgs(cmd *cobra.Command, args []string) execCmdArgs {
+	execArgs := execCmdArgs{
+		execRef:              "",
+		sshConnectionOptions: config.SSHConnectionOptions,
 	}
-	url := strings.TrimSuffix(strings.TrimSpace(stdout.String()), ".git")
-	return url, nil
-}
 
-func gitAdd() error {
-	var stdout, stderr bytes.Buffer
-	cmd := exec.Command("git", "add", ".")
-	cmd.Stdout = &stdout
-	cmd.Stderr = &stderr
+	doubleDashIdx := cmd.ArgsLenAtDash()
+	argsBeforeDoubleDash := []string{}
 
-	if err := cmd.Run(); err != nil {
-		return fmt.Errorf("failed to add: %s %s", stderr.String(), stdout.String())
+	if doubleDashIdx == -1 {
+		const errMsg = "❌ Invalid arguments. You must pass the -- flag. " +
```

Comment:
This is probably fine for now but I don't necessarily need to constrain the user to always use `--`. If they don't have any flags to pass to their sub command, they should be able to use `unweave exec python script.py`. This also looks nicer in demos. I believe Docker, Doppler, etc also do this.

Reply:
Yea, I agree that it's visually much more pleasing to see the `unweave exec python script.py`, and you're right docker does it like this. 

I based this on `kube`: 
From the `kubectl exec -h` printout:
```
Examples:
  # Get output from running the 'date' command from pod mypod, using the first container by default
  kubectl exec mypod -- date
```

I don't have a concrete example for when `--` is required, but I can imagine there's a case where we might confuse our flags with the user's flags. I'm guessing that this `--` separator will protect us from the potential difficulty in the future. And IMO it's a relatively small cost of the cli user (?). Thoughts?

Reply:
I think this is good to merge for now though. We can remove this check later on if needed